backend/main.py

```diff
@@ -224,25 +229,153 @@ def get(self):
         
         return {'files': sorted(list(children))}
 
+class DeployResource(Resource):
+    def post(self):
+        """Upload and extract a zip file to the deploy directory"""
+        if 'file' not in request.files:
+            return {'error': 'No file provided'}, 400
+        
+        file = request.files['file']
+        
+        if file.filename == '':
+            return {'error': 'No file selected'}, 400
+        
+        if not file.filename.endswith('.zip'):
+            return {'error': 'Only zip files are allowed'}, 400
+        
+        try:
+            # Create deploy directory if it doesn't exist
+            deploy_dir = os.path.join(basedir, 'deploy')
+            
+            # Clear existing deploy directory
+            if os.path.exists(deploy_dir):
+                shutil.rmtree(deploy_dir)
+            os.makedirs(deploy_dir)
+            
+            # Save the zip file temporarily
+            temp_zip_path = os.path.join(basedir, 'temp_deploy.zip')
+            file.save(temp_zip_path)
+            
+            # Extract the zip file
+            with zipfile.ZipFile(temp_zip_path, 'r') as zip_ref:
+                zip_ref.extractall(deploy_dir)
+            
+            # Remove the temporary zip file
+            os.remove(temp_zip_path)
+            
+            # List extracted files
+            extracted_files = []
+            for root, dirs, files in os.walk(deploy_dir):
+                for filename in files:
+                    rel_path = os.path.relpath(os.path.join(root, filename), deploy_dir)
+                    extracted_files.append(rel_path)
+            
+            return {
+                'message': 'Deployment successful',
+                'deploy_directory': deploy_dir,
+                'files_extracted': len(extracted_files),
+                'files': extracted_files[:20]  # Show first 20 files
+            }
+        except zipfile.BadZipFile:
+            return {'error': 'Invalid zip file'}, 400
+        except Exception as e:
+            return {'error': f'Deployment failed: {str(e)}'}, 500
+
 # Register API routes
 # Storage API routes (more specific routes first)
 api.add_resource(StorageEntryResource, '/entries/<path:entry_key>')
 api.add_resource(StorageFileRenameResource, '/storage/rename')
 api.add_resource(StorageRootResource, '/storage/')
 api.add_resource(StorageResource, '/storage/<path:path>')
+api.add_resource(DeployResource, '/deploy')
 
-@app.route('/')
-def index():
+# API health check endpoint to distinguish from static file serving
+@app.route('/api/status')
```

Comment:
Maybe call this statusz.
Look up z pages or "statusz" if you are not familiar. I found an example here: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/4827-component-statusz/README.md#sample-response-1